src/v3.ts

```diff
@@ -154,27 +153,107 @@ export default function generateTypesV3(
+        if (responses.description) output += comment(responses.description);
+        output += `"${method}": {\n`;
+
+        // handle parameters
+        if (responses.parameters) {
+          output += `parameters: {\n`;
+          const allParameters: Record<
+            string,
+            Record<string, OpenAPI3Parameter>
+          > = {};
+          responses.parameters.forEach((p) => {
+            if (!allParameters[p.in]) allParameters[p.in] = {};
+            // TODO: handle $ref parameters
+            if (p.name) {
+              allParameters[p.in][p.name] = p;
+            }
+          });
+
+          Object.entries(allParameters).forEach(([loc, locParams]) => {
+            output += `"${loc}": {\n`;
+            Object.entries(locParams).forEach(([paramName, paramProps]) => {
+              if (paramProps.description)
+                output += comment(paramProps.description);
+              output += `"${paramName}"${
+                paramProps.required === true ? "" : "?"
+              }: ${transform(paramProps.schema)};\n`;
+            });
+            output += `}\n`;
+          });
+          output += `}\n`;
+        }
 
+        // handle responses
+        output += `responses: {\n`;
+        Object.entries(responses.responses).forEach(
+          ([statusCode, response]) => {
+            if (response.description) output += comment(response.description);
+            if (!response.content || !Object.keys(response.content).length) {
+              output += `"${statusCode}": any;\n`;
+              return;
+            }
+            output += `"${statusCode}": {\n`;
+            Object.entries(response.content).forEach(
+              ([contentType, encodedResponse]) => {
+                output += `"${contentType}": ${transform(
+                  encodedResponse.schema
+                )};\n`;
+              }
+            );
+            output += `}\n`;
+          }
+        );
+        output += `}\n`;
+        output += `}\n`;
+      });
+      output += `}\n`;
+    });
+    return output;
+  }
+
+  if (rawSchema) {
     return `export interface schemas {
-      ${schemas}
-    }`;
+  ${createKeys(propertyMapped, Object.keys(propertyMapped))}
+}`;
   }
 
-  const schemas = `schemas: {
-    ${createKeys(propertyMapped, Object.keys(propertyMapped))}
-  }`;
-
-  const responses = !components.responses
-    ? ``
-    : `responses: {
-    ${createKeys(components.responses, Object.keys(components.responses))}
-  }`;
-
-  // note: make sure that base-level schemas are required
-  return `export interface components {
-    ${schemas}
-    ${responses}
-  }`;
+  // now put everything together
+  let finalOutput = "";
+
+  // handle paths
+  if (Object.keys(paths).length) {
+    finalOutput += `export interface paths {
+  ${transformPaths(paths)}
+}
+
+`;
+  }
+
+  finalOutput += "export interface components {\n";
+
+  // TODO: handle components.parameters
```

Comment:
We also don’t yet handle `components.parameters`, which would be needed for `$ref` parameters (above)